logistics/scripts/activities/build/CloudOds/PrepareDatabasesForExport.ps1

```diff
@@ -354,69 +427,79 @@ $deploymentTasks = @(
         Script = {
-            $csb = $script:postgreSQLConnectionStrings[$databaseIds.admin.connectionStringKey]
+            $databaseType = $postgreSQLConfig.ApiSettings.DatabaseTypes.Admin
+            $connectionStringKey = $postgreSQLConfig.ApiSettings.ConnectionStringKeys[$databaseType]
+            $csb = $postgreSQLConfig.ApiSettings.csbs[$connectionStringKey]
             $params = @{
                 databaseConnectionString = $csb
                 databaseBackupName       = $csb["Database"]
                 backupDirectory          = $artifactPath
                 multipleBackups          = $true
-                engine                   = "PostgreSQL"
+                engine                   = $postgreSQLConfig.ApiSettings.Engine
             }
             Backup-DatabaseTemplate $params
         }
     }
     @{
         Name   = "Export Security Database from PostgreSQL as .sql"
         Script = {
-            $csb = $script:postgreSQLConnectionStrings[$databaseIds.security.connectionStringKey]
+            $databaseType = $postgreSQLConfig.ApiSettings.DatabaseTypes.Security
+            $connectionStringKey = $postgreSQLConfig.ApiSettings.ConnectionStringKeys[$databaseType]
+            $csb = $postgreSQLConfig.ApiSettings.csbs[$connectionStringKey]
             $params = @{
                 databaseConnectionString = $csb
                 databaseBackupName       = $csb["Database"]
                 backupDirectory          = $artifactPath
                 multipleBackups          = $true
-                engine                   = "PostgreSQL"
+                engine                   = $postgreSQLConfig.ApiSettings.Engine
             }
             Backup-DatabaseTemplate $params
         }
     }
     @{
         Name   = "Export Empty Template Database from PostgreSQL as .sql"
         Script = {
-            $csb = Get-DbConnectionStringBuilderFromTemplate -templateCSB $script:postgreSQLConnectionStrings[$databaseIds.ods.connectionStringKey] -replacementTokens "Ods"
+            $databaseType = $postgreSQLConfig.ApiSettings.DatabaseTypes.Ods
+            $connectionStringKey = $postgreSQLConfig.ApiSettings.ConnectionStringKeys[$databaseType]
+            $csb = Get-DbConnectionStringBuilderFromTemplate -templateCSB $postgreSQLConfig.ApiSettings.csbs[$connectionStringKey] -replacementTokens 'Ods_Empty_Template'
             $params = @{
-                databaseConnectionString = Get-DbConnectionStringBuilderFromTemplate -templateCSB $script:postgreSQLConnectionStrings[$databaseIds.ods.connectionStringKey] -replacementTokens "Ods_Empty_Template"
+                databaseConnectionString = $csb
                 databaseBackupName       = $csb["Database"]
                 backupDirectory          = $artifactPath
                 multipleBackups          = $true
-                engine                   = "PostgreSQL"
+                engine                   = $postgreSQLConfig.ApiSettings.Engine
```

Comment:
This needs to create the shared instance database instead of the empty. You can see before that it was actually creating one csb with "Ods" and then settings the databaseConnectionString variable with the csb of the empty.

Reply:
done